### PR TITLE
also return classes when caling goToDef on a constructor call

### PR DESCRIPTION
**Pull request by iisaduan**
When requesting definition on a non-extended class,
```typescript
class Foo {
    /*
     * possibly a lot of code
     */
    constructor()
}
const foo = new /*goToDefinition*/Foo()
```
we now also return the class definition, in addition to the constructor.

Described here/generated by/inspired by https://github.com/microsoft/TypeScript/issues/59241#issuecomment-2248737599 (related #59241)

### CODE REVIEW

**Comment by iisaduan** [comment]
@typescript-bot pack this

File: src/services/goToDefinition.ts [hunk 1]
```diff
@@ -230,14 +231,16 @@ export function getDefinitionAtPosition(program: Program, sourceFile: SourceFile
         const sigInfo = createDefinitionFromSignatureDeclaration(typeChecker, calledDeclaration, failedAliasResolution);
         // For a function, if this is the original function definition, return just sigInfo.
         // If this is the original constructor definition, parent is the class.
+        let includeDeclarations: ((d: Declaration) => boolean) | undefined;
```
Comment (RyanCavanaugh) [inline src/services/goToDefinition.ts:234]: This is named like it's a boolean; I would suggest something like `declarationFilter`

File: src/services/goToDefinition.ts [hunk 2]
```diff
@@ -584,9 +587,10 @@ function isExpandoDeclaration(node: Declaration): boolean {
     return !!containingAssignment && getAssignmentDeclarationKind(containingAssignment) === AssignmentDeclarationKind.Property;
 }
 
-function getDefinitionFromSymbol(typeChecker: TypeChecker, symbol: Symbol, node: Node, failedAliasResolution?: boolean, excludeDeclaration?: Node): DefinitionInfo[] | undefined {
-    const filteredDeclarations = filter(symbol.declarations, d => d !== excludeDeclaration);
-    const signatureDefinition = getConstructSignatureDefinition() || getCallSignatureDefinition();
+function getDefinitionFromSymbol(typeChecker: TypeChecker, symbol: Symbol, node: Node, failedAliasResolution?: boolean, excludeDeclaration?: Node, includeDeclarations?: (d: Declaration) => boolean): DefinitionInfo[] | undefined {
```
Comment (RyanCavanaugh) [inline src/services/goToDefinition.ts:590]: There's only one inbound call to this function that uses `excludeDeclaration`; let's just simplify this to take a predicate and then that call can pass `e => e !== node`

**Review by RyanCavanaugh — APPROVED** [review]
LGTM after a couple nits

**Pull request merged**